### PR TITLE
Fix/etp 51 redo

### PR DESCRIPTION
**Pull request by cliffordp**
[ETP-51]

[ETP-51]: https://moderntribe.atlassian.net/browse/ETP-51

### CODE REVIEW

File: src/template-tags/tickets.php [hunk 1]
```diff
@@ -403,6 +396,48 @@ function tribe_events_ticket_is_on_sale( Tribe__Tickets__Ticket_Object $ticket )
...
+
+			// Skip tickets that are for a different provider than the event provider.
+			if (
+				$default_provider !== $ticket_provider->class_name
+				&& Tribe__Tickets__RSVP::class !== $ticket_provider->class_name
+			) {
+				continue;
+			}
+
+			$has_tickets_available = ( $has_tickets_available || tribe_events_ticket_is_on_sale( $ticket ) );
```
Comment (Camwyn) [inline src/template-tags/tickets.php]: This is going to run through all the tickets no matter what, but it looks like you can break out once this is set to true?

Comment (Camwyn) [reply src/template-tags/tickets.php]:
```suggestion
			if ( tribe_events_ticket_is_on_sale( $ticket ) ) {
			    $has_tickets_available = true;
			    break;
			}
```

You could even shorten it a bit more by leaving out `$has_tickets_available` and just setting `$ticket_availability[ $post_id ]` ?

**Comment by Camwyn** [comment]
Can we get a link to the original PR in the description - assuming there is more info there?
Or a description, etc.
Just a ticket number link is never enough 😞

**Comment by cliffordp** [comment]
@Camwyn well Jira auto-connects them all so I thought the info there would be fine. Nevertheless, for easier reference ;)
* https://github.com/moderntribe/event-tickets/pull/1523
* https://github.com/moderntribe/event-tickets/pull/1532
* https://github.com/moderntribe/event-tickets/pull/1533

File: src/template-tags/tickets.php (hunk 1 above)
Comment (cliffordp) [reply src/template-tags/tickets.php]: `tribe_tickets_is_current_time_in_date_window()` came from @borkweb in https://github.com/moderntribe/event-tickets/pull/1533/commits/e21e6c2fd8769983ddb72959b1030784c469003a so I didn't want to affect it for eventual merging with B20.01/G20.01

But good suggestions so let's do it anyway :)

Comment (sc0ttkclark) [reply src/template-tags/tickets.php]: I added a note for myself to come back to this function in my optimization tasks. We can keep as is for now.

**Pull request merged**